src/compiler/utilities.ts

```diff
@@ -3543,6 +3543,18 @@ namespace ts {
         }
     }
 
+    /**
+     * Find the index where arrayA and arrayB differ
+     */
+    export function findDiffIndex<T>(arrayA: ReadonlyArray<T>, arrayB: ReadonlyArray<T>) {

```

Comment:
It would make more sense for this function to return `undefined` on failure, rather than an invalid index.